scapy/layers/inet6.py

```diff
@@ -2053,6 +2054,39 @@ class ICMPv6NDOptDNSSL(_ICMPv6NDGuessPayload, Packet):  # RFC 6106
     def mysummary(self):
         return self.sprintf("%name% ") + ", ".join(self.searchlist)
 
+
+# URI MUST be padded with NUL (0x00) to make the total option length
+# (including the Type and Length fields) a multiple of 8 bytes.
+# https://www.rfc-editor.org/rfc/rfc8910.html#name-the-captive-portal-ipv6-ra-
+class CaptivePortalURI(StrLenField):
```

Comment:
I wonder if this can be moved into fields.py and renamed StrLenNullField

Reply:
I think this field is kind of specific in the sense that it's weird and unlikely to be used anywhere else so by analogy with other ND-related fields (like `DomainNameListField`) that have to deal with that sort of padding I thought it would be better to keep it close to the only place where it's currently used.

Reply:
ACK. Let's keep it where it is.